TabularClassifier.ipynb

```diff
@@ -0,0 +1,207 @@
+    "\n",
+    "    # the last column is the label\n",
+    "    column_datatypes.append(\"label\")\n",
+    "\n",
+    "    # TabularClassifier assumes no header so we add the first label \n",
+    "    # (considered a column name in pandas) as part of the test labels\n",
+    "    test_labels = [df.columns[-1]] + list(df[df.columns[-1]])\n",
+    "\n",
+    "    return n_classes, column_datatypes, test_labels\n",
+    "\n",
+    "\n",
+    "download_census_income_dataset()\n",
+    "reformat_test_csv()\n",
+    "(n_classes, column_datatypes, test_labels) = get_dataset_metadata()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Defining the TabularClassifier model\n",
+    "\n",
+    "### Arguments\n",
+    "- `model_size`: Controls how big the model is. Options are ‘small’, ‘medium’,  ‘large’, or a target model size in gigabytes, i.e. “4 Gb” or “4Gb”.\n",
+    "- `n_classes`: How many output classes are in the dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "classifier = bolt.TabularClassifier(model_size=\"medium\", n_classes=n_classes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 2: Training the TabularClassifier\n",
+    "\n",
+    "### Arguments\n",
+    "- `train_file`: The dataset to train on. The expected format is a csv file with comma separated columns. This csv file should not have a header. The TabularClassifier also does not attempt to shuffle the dataset, the user should shuffle the dataset before calling train if necessary.\n",
+    "- `column_datatypes`: List of column types. Should be a list of strings with the types of the columns, which are one of: \"numeric\", \"categorical\", or \"label\". There should be only one \"label\" column. \n",
+    "- `epochs`: This determines the number of epochs to train on. If there is 1 epoch then it uses a streaming dataset, otherwise it loads the dataset into memory and trains for the specified number of epochs.\n",
+    "- `learning_rate`: Determines the learning_rate for training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "classifier.train(\n",
+    "    train_file=TRAIN_FILE,\n",
+    "    column_datatypes=column_datatypes,\n",
+    "    epochs=1,\n",
+    "    learning_rate=0.01,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 3: Testing The Classifier\n",
+    "### Arguments\n",
+    "- `test_file`: The test dataset to run. The file should follow the same format as the test file, column datatypes are assumed to be identical.\n",
+    "- `output_file`: This is an optional parameter. If it is specified then as the data is processed it writes the names of the predicted classes into this file. Each predicted class name is on its own line."
```

Comment:
You should also say what happens if it isn't specified. Also, it seems like a problem that you need to pass in labels. How does the customer know we aren't just cheating them and saying we get 90% accuracy or whatever? How are you supposed to use the classifier in production?

Reply:
yep :) this is something we should've done from the beginning

right now it doesn't seem like a huge huge deal since this isn't in production yet. But yes this is definitely on the round for next steps